### PR TITLE
Stack and queue

### PR DESCRIPTION
**Pull request by JamesDansie**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/stacksandqueues/Stack.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class Stack<E> {
+
+    Node<E> top;
+
+    //adds a node at the beginning
+    public void push(E o){this.top = new Node<E>(o, this.top);}
+
+    public E pop() {
+        if(top == null){
+            throw new IllegalArgumentException("The stack is empty");
```
Comment (ncarignan) [inline code401challenges/src/main/java/code401challenges/stacksandqueues/Stack.java]: this could be EmptyStackException

File: code401challenges/src/main/java/code401challenges/stacksandqueues/Stack.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package code401challenges.stacksandqueues;
+
+public class Stack<E> {
+
+    Node<E> top;
+
+    //adds a node at the beginning
+    public void push(E o){this.top = new Node<E>(o, this.top);}
```
Comment (ncarignan) [inline code401challenges/src/main/java/code401challenges/stacksandqueues/Stack.java]: nice job implementing this clean version

File: code401challenges/src/test/java/code401challenges/stacksandqueues/StackTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                stack.peek());
+    }
+
+    @Test
+    public void testEmpty(){
+        Stack<String> empty = new Stack<>();
+        assertNull(empty.top);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```
Comment (ncarignan) [inline code401challenges/src/test/java/code401challenges/stacksandqueues/StackTest.java]: This is a good test to include on all the methods that throw this error
